lock-api/src/main/java/com/palantir/lock/client/TransactionStarter.java

```diff
@@ -79,6 +80,21 @@ StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
         }
     }
 
+    List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTransactionBatch(int count) {
+        return autobatcher.applyBatch(
+                IntStream.range(0, count).mapToObj($ -> (Void) null).collect(Collectors.toList()))
+                .stream()
+                .map(result -> {
+                    try {
+                        return result.get();
+                    } catch (ExecutionException e) {
+                        throw Throwables.throwUncheckedException(e.getCause());
```

Comment:
Again, I think you can't just throw exceptions here: @gmaretic can correct me, but I think timelock can 1. choose to return you less transactions than you asked for. 2. Dunno if the disruptor can guarantee that all your requested number of transactions will end up in a single batch to timelock.

Therefore, I think you should assume that each of those futures can fail independently, therefore you need to decide how to handle that. At the very least you need to make sure that you do not leak resources: StartIdentifiedAtlasDbTransactionResponse contains the immutable timestamp lock (again someone more versed in this can explain what happens when that is leaked), which in the #wrap method is explicitly unlocked when something fails.